model/deck.go

```diff
@@ -36,11 +42,37 @@ func NewDeck() Deck {
 	return &d
 }
 
+func newDeckWithDealt(dealt map[Card]struct{}) Deck {
+	d := newDeck()
+	if len(dealt) == 0 {
+		return d
+	}
+
+	for i := 0; i < len(d.cards); i++ {
+		c := d.cards[i]
+		lastValidCard := 51 - d.numDealt
+		if i > lastValidCard {
+			break
+		}
+		if _, ok := dealt[c]; ok {
+			tmp := d.cards[lastValidCard]
+			d.cards[lastValidCard] = d.cards[i]
+			d.cards[i] = tmp
```

Comment:
Make this more concise:
`d.cards[lastValidCard], d.cards[i] = d.cards[i], d.cards[lastValidCard]`

Reply:
I wish -- for some reason, it was causing me errors. I can't seem to repro now, but I wanna leave it as is, cuz this is boring and it makes sense